build-system/babel-config/minified-config.js

```diff
@@ -68,17 +68,22 @@ function getMinifiedConfig() {
       shippedProposals: true,
     },
   ];
+  const presetTypescript = [
+    '@babel/preset-typescript',
+    {jsxPragma: 'Preact', jsxPragmaFrag: 'Preact.Fragment'},
+  ];
 
   return {
     compact: false,
     plugins,
     sourceMaps: true,
-    presets: [presetEnv],
+    presets: [presetTypescript, presetEnv],
     retainLines: true,
     assumptions: {
       constantSuper: true,
       noClassCalls: true,
       setClassMethods: true,
+      setPublicClassFields: true,
```

Comment:
I'm worried about adding this, because we do sometimes have setters. If a setter is defined in the base-class, this will lead to improper behavior that will prevent us from removing the class-fields transform later:

```js
class Base {
  set foo(v) { throw new Error("shouldn't be called") }
}

class Sub {
  foo = 1;
}
```

Can we add a linter for this?

Reply:
The [documentation](https://babeljs.io/docs/en/assumptions#setpublicclassfields) specifically calls out shadowing getters as an issue. Are you sure it also applies to setter? If so we should also update the documentation

> When using public class fields, assume that they don't shadow any getter in the current class, in its subclasses or in its superclass. Thus, it's safe to assign them rather than using Object.defineProperty.

Reply:
It applies to accessors. Setters are more worrying, because people use them to initialize values from the subclass that are then usable during the superclass's constructor phase. This is invalid in real JS. See https://github.com/lit/lit/issues/2446#issuecomment-1021501005 for a related issue.

Reply:
I had a hunch that TS might take care of this for us, and it does. That said, it would only help out with TS-checked code.

<img width="857" alt="Screen Shot 2022-01-26 at 12 25 04 AM" src="https://user-images.githubusercontent.com/4656974/151108565-7d21f1e5-d3ae-424c-a062-22fd4d6d4b07.png">